src/cpp/session/modules/SessionRCompletions.R

```diff
@@ -2729,7 +2735,8 @@ assign(x = ".rs.acCompletionTypes",
          if (length(objectNames))
          {
             completions <- .rs.selectFuzzyMatches(objectNames, token)
-            if (inherits(object, "data.frame"))
+
+            if (inherits(object, c("data.frame")) || (isNamespaceLoaded("arrow") && inherits(object, c("ArrowTabular", "Dataset"))))
```

Comment:
I don't love this, I wonder if we could have a generic that says if an object is a rectangle/columnar/tidy ?

Reply:
Need this too because `ds |> something_dplyr()` returns an `arrow_dplyr_query` object, which also has `names()` available that you might want to tab complete.

In case it's relevant, `arrow_dplyr_query` is S3 not R6, and `names()` returns the column names.

```suggestion
            if (inherits(object, c("data.frame")) || (isNamespaceLoaded("arrow") && inherits(object, c("ArrowTabular", "Dataset", "arrow_dplyr_query"))))
```